src/oc_erchef/apps/oc_chef_wm/src/oc_chef_wm_keys.erl

```diff
@@ -114,30 +159,91 @@ auth_info(Req, #base_state{resource_args = TargetType, resource_state = #key_sta
 auth_info(Req, #base_state{ resource_mod = Mod } = State) ->
     Mod:auth_info(wrq:method(Req), Req, State).
 
-auth_info('GET', Req, #base_state{resource_state = #key_state{parent_authz_id = AuthzId}}= State) ->
-    {{actor, AuthzId}, Req, State};
+-spec auth_info(list(), wm_req(), chef_wm:base_state()) -> 
+                       {{halt, 403} | {actor, object_id()} | {actor, object_id(), update} | authorized,
+                        wm_req(), chef_wm:base_state()}.
+%% GET /users/:user/keys(/:key)
+auth_info('GET', Req, #base_state{organization_guid = undefined,
+                                  resource_state = #key_state{parent_authz_id = ActorId}}= State) ->
+    {{actor, ActorId}, Req, State};
+%% GET /orgs/:org/clients/:client/keys(/:key) OR
+%% GET /orgs/:org/users/:user/keys(/:key)
+auth_info('GET', Req, State) ->
+    read_auth_info(Req, State);
+%% Non GET methods
 auth_info(_Method, Req, #base_state{resource_state = #key_state{parent_authz_id = AuthzId}}= State) ->
     {{actor, AuthzId, update}, Req, State}.
 
+%% @private
+
+%% This code serves 4 endpoints:
+%% + /orgs/:org/clients/:client/keys(/:key)
+%% + /orgs/:org/users/:user/keys(/:key)
+%% For /users/:user/keys(/:key), we obey what is outlined in auth_info above.
+%% For the org scoped ones, we want to check if the requestor is a member
+%% of <org>_actor_keys_access_group on GET requests to give all users and clients
+%% access on all users and clients if they share an org.
+-spec read_auth_info(wm_req(), chef_wm:base_state()) ->
+                            {{actor, object_id()}, wm_req(), chef_wm:base_state()} |
+                            {authorized, wm_req(), chef_wm:base_state()} |
+                            {{halt, 403}, wm_req(), chef_wm:base_state()}.
+read_auth_info(Req, #base_state{organization_name = OrgName,
+                                chef_authz_context = AuthzContext,
+                                resource_state = #key_state{parent_authz_id = ActorId}} = State) ->
+    %% This is the atom true or false, return the oppsite to trigger
+    %% a 403 if this actor is not a member of <org>_actor_keys_access_group.
+    GroupName = binary_to_list(OrgName) ++ "_actor_keys_access_group",
+    Group = oc_chef_authz_db:fetch_global_group(AuthzContext, GroupName),
+    GroupId = Group#oc_chef_group.authz_id,
+    case oc_chef_authz:is_actor_transitive_member_of_group(oc_chef_authz:superuser_id(), ActorId, GroupId) of
+        {error, not_found} ->
+            %% Let oc_chef_wm_base:forbidden sort things out, it handles 404s.
+            {{actor, ActorId}, Req, State};
+        true ->
+            {authorized, Req, State};
+        false ->
+            {{halt, 403}, Req, State}
+    end.
```

Comment:
I'm wondering if it makes more sense to implement this in oc_chef_wm_base so that future endpoints can use the same mechanism.  `auth_info` in this module could return `{{member_of, GroupName}, Req, State}` and this logic could be moved into the oc_chef_wm_base:forbidden (or some function it calls).